RickLeinecker/Fortuna
# Marketplace API implementation

**Description**
Marketplace API implemented. Much of how to use them I've documented in comments on the files, so I won't reiterate them here.

The most notable things are the changes to the User model, so be wary of anything that might be using it (mostly regarding APIs that have been using the User model).

Another thing to note is that I included implementation that supports casusBlocks, despite use not supporting the casusBlock inventory. It's mostly for future expansion. 

Resolves #93 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
Passes Flow.
I did testing locally, but it was a pain to reproduce since we have no test entries in the database, so it took a while. But to sum it up if you want to try it:
1. Checkout branch
2. `npm install`
3. `npm run build`
4. Have the following things in the DB
    4a. Two users with currency and inventory manually added to them
    4b. One tank associated with one the users
5. Create Sale Post with one user (list tank or inventory items)
6. Have other user do a transaction of that sale.
7. Verify the transfer of item(s)

**Final Checklist:**
Go down the list and check them off.
